StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1135UsingDirectivesMustBeQualified.cs

```diff
@@ -124,37 +125,114 @@ private static void CheckUsingDeclaration(SyntaxNodeAnalysisContext context, Usi
+                    }
+
+                    AppendCanonicalString(builder, typeArgumentList.Arguments[i]);
                 }
-            }
 
-            return StringBuilderPool.ReturnAndFree(builder);
+                builder.Append(">");
+                return true;
+
+            case QualifiedNameSyntax qualifiedName:
+                AppendCanonicalString(builder, qualifiedName.Left);
+                builder.Append(".");
+                AppendCanonicalString(builder, qualifiedName.Right);
+                return true;
+
+            case PredefinedTypeSyntax predefinedType:
+                builder.Append(predefinedType.Keyword.Text);
+                return true;
+
+            case ArrayTypeSyntax arrayType:
+                AppendCanonicalString(builder, arrayType.ElementType);
+                foreach (var rankSpecifier in arrayType.RankSpecifiers)
+                {
+                    builder.Append("[");
+                    builder.Append(',', rankSpecifier.Rank - 1);
+                    builder.Append("]");
+                }
+
+                return true;
+
+            case NullableTypeSyntax nullableType:
+                AppendCanonicalString(builder, nullableType.ElementType);
+                builder.Append("?");
+                return true;
+
+            case OmittedTypeArgumentSyntax _:
+                return false;
+
+            default:
+                if (TupleTypeSyntaxWrapper.IsInstance(type))
+                {
+                    var tupleType = (TupleTypeSyntaxWrapper)type;
+
+                    builder.Append("(");
+
+                    var elements = tupleType.Elements;
+                    for (int i = 0; i < elements.Count; i++)
+                    {
+                        if (i > 0)
+                        {
+                            builder.Append(", ");
+                        }
+
+                        AppendCanonicalString(builder, elements[i].Type);
+                        if (!elements[i].Identifier.IsKind(SyntaxKind.None))
+                        {
+                            builder.Append(" ").Append(elements[i].Identifier.Text);
+                        }
+                    }
+
+                    builder.Append(")");
+                    return true;
+                }
+                else if (RefTypeSyntaxWrapper.IsInstance(type))
+                {
+                    var refType = (RefTypeSyntaxWrapper)type;
+                    builder.Append(refType.RefKeyword.Text);
+                    if (refType.ReadOnlyKeyword.IsKind(SyntaxKind.ReadOnlyKeyword))
+                    {
+                        builder.Append(" ").Append(refType.ReadOnlyKeyword.Text);
+                    }
+
+                    builder.Append(" ");
+                    AppendCanonicalString(builder, refType.Type);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
```

Comment:
❓ Would the runtime performance of `switch (type.Kind())` be better? It would definitely improve readability, as the tuple type and ref type no longer will have to handled in the default case.

Reply:
Fixed. Also added a bunch of tests which revealed more bugs.